playbooks/tasks/portal-role-task-mongo-replicaset-get-primary.yml

```
@@ -3,25 +3,28 @@
 # Find primary node in the MongoDB replicaset
 # TODO: This ansible playbook task, should be moved to portal role
 
-# Get mongo cluster hosts
-- name: Get mongo cluster hosts
-  set_fact:
-    mongo_cluster_hosts: "{{ (mongo_cluster_hosts | default([])) + [item] }}"
-  when: >-
-    hostvars[item].portal_cluster_id is defined and
-    hostvars[item].portal_cluster_id == portal_cluster_id
-  loop: "{{ groups['webportals'] }}"
-  delegate_to: localhost
+# This task expects:
+# - MongoDB node where we run this task is a member of the replicaset
 
 - name: Reset mongo primary host
   set_fact:
     mongo_primary_host: ""
 
-- name: Include getting mongo replicaset primary member
-  include_tasks: tasks/portal-role-task-mongo-replicaset-get-primary-sub-task.yml
-  loop: "{{ mongo_cluster_hosts }}"
-  loop_control:
-    loop_var: "host_item"
+- name: Get MongoDB replicaset primary member
+  include_tasks: tasks/portal-role-task-mongo-shell-eval.yml
+  vars:
+    mongodb_shell:
+      # Due to yaml escaping following eval must use ':', not ":".
+      # isMaster.primary returns string in format '<FQDN>:<port>', we need to
+      # extract only the first subdomain which is Ansible host.
+      eval: |
+        rs.isMaster().primary.split('.')[0]
```

based on the comment shouldn't it be this:
```suggestion
        rs.isMaster().primary.split(':')[0]
```

---

That suggestion is not correct.

`rs.isMaster().primary` returns e.g. `us-ny-1.siasky.net:27017`.
We need to extract `Ansible host` which is just `us-ny-1` (not FQDN `us-ny-1.siasky.net`).

---

I have updated the comment not to mention `FQDN` but subdomain.

---

I guess I'm not following the comment. What is it referring to? Where do we need to be referring to `':'`? 

Right now the comment s more confusing the helpful IMO.

---

oh, I have removed the complete comment `# Due to yaml escaping following eval must use ':', not ":".`
because the eval scripts has changed since the comment was added, the eval script doesn't contain `:` character anymore.